File: common/security/src/main/java/org/yechan/AuthTokenProperties.java
package org.yechan;

import jakarta.validation.constraints.Min;
import jakarta.validation.constraints.NotBlank;
import org.springframework.boot.context.properties.ConfigurationProperties;
import org.springframework.validation.annotation.Validated;

@Validated
@ConfigurationProperties(prefix = "auth.token")
public record AuthTokenProperties(
    @NotBlank String salt,
    @Min(1) long accessExpiresIn,
    @Min(1) long refreshExpiresIn
) {
}

File: common/security/src/main/kotlin/org/yechan/AuthTokenProperties.kt
package org.yechan

import jakarta.validation.constraints.Min
import jakarta.validation.constraints.NotBlank
import org.springframework.boot.context.properties.ConfigurationProperties
import org.springframework.validation.annotation.Validated

@ConfigurationProperties(prefix = "auth.token")
@Validated
data class AuthTokenProperties(
    @field:NotBlank
    val salt: String,
    @field:Min(1)
    val accessExpiresIn: Long,
    @field:Min(1)
    val refreshExpiresIn: Long,
)
